### ISSUE TITLE
Missing Test Case - Find First and Last Position of Element in Sorted Array

### ISSUE THREAD

**Issue by pratikdey70** [LeetCode-Feedback/LeetCode-Feedback#25339]
### LeetCode Username

Pratik Dey

### Problem Number, Title, and Link

https://leetcode.com/problems/find-first-and-last-position-of-element-in-sorted-array

### Bug Category

Missing test case *(Incorrect/Inefficient Code getting accepted because of missing test cases)*

### Bug Description

All testcases are provided but there should be some testcase like
e.g. [2,3,4,4,4,5,6,6,6,6,6,8]  target=6
All the test cases repeating element two times need to check whether its working fine for more than 2 repeating number or not because with 2 time repeat it becomes 2nd pos of an element as well

### Language Used for Code

C++

### Code used for Submit/Run operation

```bash
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int n = nums.size();
        int firstpos=-1;
        int secondpos=-1;
        for(int i=0; i<n ; i++){
            if(nums[i]==target){
                if(firstpos==-1){
                    firstpos = i;
                }
                secondpos = i;
            }
        }
        return {firstpos,secondpos};
    }
};
```

### Expected behavior

Expecting that code will properly give the last position of occurrence of an element but in this question there were no such testcase for testing  if there are multiple occurrences, but we don't update the last occurrence properly.

**Comment by pratikdey70** [comment]
> LeetCode Support commented: Hello,
> 
> Thank you for contributing to LeetCode and submitting a missing test case. Upon review, we found that your provided test case did not cause your solution code to fail and it was Accepted as before. This indicates that your test case may not address the specific edge case or scenario needed to highlight the issue in your solution.
> 
> Here are the test cases you submitted:
> 
> Test case #1:
> 
> ```
> [2,3,4,4,4,5,6,6,6,6,6,8]
> 6
> ```
> 
> We encourage you to review your code and consider other potential edge cases that might reveal the underlying problem. If you identify any additional test cases or have further questions, please feel free to reply to this message or open a new issue with us.
> 
> We appreciate your engagement and efforts in helping us enhance our platform.
> 
> Best regards, LeetCode Support Team

> Very Large Arrays:
Scenario: If the array is very large (e.g., millions of elements), the time complexity of the function is O(n) where n is the length of the array. This should still work in reasonable time for large arrays as long as there’s sufficient memory.
Problem: In the case of very large arrays, performance might degrade if you're working with a massive array and need to optimize further, but this is already linear complexity.
Expected Output: Correct range of first and last positions.

>Target Occurs Only Once:
Scenario: If the target element appears only once in the array, both firstpos and secondpos should be the same value.
Problem: The current implementation correctly handles this by setting both firstpos and secondpos to the same index when the target occurs once.
